cairo/tests/conftest.py

```diff
@@ -60,16 +134,29 @@ def pytest_configure(config):
 
     from tests.utils.args_gen import Environment, Evm, Message, MessageCallOutput
 
+    init_tracer()
+
     # Apply patches at module level before any tests run
     ethereum.cancun.vm.Evm = Evm
     ethereum.cancun.vm.Message = Message
     ethereum.cancun.vm.Environment = Environment
     ethereum.cancun.vm.interpreter.MessageCallOutput = MessageCallOutput
 
-    # Mock the Extended type because hypothesis cannot handle the RLP Protocol
-    # Needs to be done before importing the types from ethereum.cancun.trie
-    # trunk-ignore(ruff/F821)
-    ethereum_rlp.rlp.Extended = Union[Sequence["Extended"], bytearray, bytes, Uint, FixedUnsigned, str, bool]  # type: ignore
+    # Mock the Extended type
+    ethereum_rlp.rlp.Extended = Union[Sequence["Extended"], bytearray, bytes, Uint, FixedUnsigned, str, bool]  # type: ignore # noqa: F821
+
+    # Patching evm_trace:
+    # - Problem: Global patches of `ethereum.trace.evm_trace` are not reflected in places where `evm_trace` is imported in EELS.
+    # - Cause: `ethereum.cancun.vm.interpreter` (and other modules) imports `evm_trace` locally (e.g., `from ethereum.trace import evm_trace`)
+    #   at module load time, caching the original `discard_evm_trace`. Patching `ethereum.trace.evm_trace` later didn’t
+    #   update this local reference due to Python’s import caching.
+    # - Solution: Explicitly patch both `ethereum.trace.evm_trace` globally and
+    #   `ethereum.cancun.vm.interpreter.evm_trace` locally (and other places where `evm_trace` is imported).
```

Comment:
still thinking that there is most probably a way to patch globally, with unittest.mock or other options. I don't mind leaving this as is, but leave a TODO to investigate other options

Reply:
probably but I could not get anything other to work